causal_automl/download_eia_metadata.py

```diff
@@ -35,303 +36,294 @@
-                "dataset_id": dataset_id_clean,
-                "name": data.get("name"),
-                "description": data.get("description"),
-                "frequency_id": frequency.get("id"),
-                "frequency_alias": frequency.get("alias"),
-                "frequency_description": frequency.get("description"),
-                "frequency_query": frequency.get("query"),
-                "frequency_format": frequency.get("format"),
-                "facets": data.get("facets"),
-                "data": metric_id,
-                "data_alias": metric_info.get("alias"),
-                "data_units": metric_info.get("units"),
-                "start_period": data.get("startPeriod"),
-                "end_period": data.get("endPeriod"),
-                "parameter_values_file": param_file_path,
-            }
-            flattened_metadata.append(metadata)
-    return flattened_metadata
-
-
-def _get_leaf_route_data(
-    root_route: str, api_key: str
-) -> Dict[str, Dict[str, Any]]:
-    """
-    Traverse the API tree and collect metadata from all leaf routes.
-
-    This function performs a breadth-first traversal over all sub-routes beginning at
-    `root_route`. For each route that has no children (i.e., a leaf), it fetches and stores
-    the associated metadata.
-
-    Example output:
-    {
-        "electricity/retail-sales": {
-            "id": "retail-sales",
-            "name": "Electricity Sales to Ultimate Customers",
-            "frequency": [...],
-            "facets": [...],
-            "data": {...},
+            "data": {
+                "revenue": {
+                    "alias": "Revenue from Sales to Ultimate Customers",
+                    "units": "million dollars"
+                },
+                ...
+            },
             "startPeriod": "2001-01",
             "endPeriod": "2025-01",
             "defaultDateFormat": "YYYY-MM",
             "defaultFrequency": "monthly"
-        },
-        ...
-    }
+        }
+
+        :param route: endpoint path like "electricity/retail-sales"
+        :return: content from the EIA API response
+        """
+        # Build the full API request URL.
+        url = f"{BASE_URL}/{route}?api_key={self._api_key}"
+        # Send HTTP GET request to the EIA API.
+        response = requests.get(url, timeout=20)
+        # Parse JSON content.
+        json_data = response.json()
+        # Get response from parsed payload.
+        data: Dict[str, Any] = {}
+        data = json_data.get("response", {})
+        return data
+
+
+    def _get_leaf_route_data(self) -> Dict[str, Dict[str, Any]]:
+        """
+        Traverse the API tree and collect metadata from all leaf routes.
+
+        This function performs a breadth-first traversal over all sub-routes beginning at
+        `root_route`. For each route that has no children (i.e., a leaf), it fetches and stores
+        the associated metadata.
+
+        Example output:
```

Comment:
You can put this after the `return` in the docstring (we inline examples of interface vars)